### DISCUSSION TITLE
How to can add Sprite2d and SimplePlane into single Sprite

### DISCUSSION THREAD

**Discussion by gapgag55** [pixijs/pixijs#8271] [Q&A] [answered]
### Expected Behavior
How to can add Sprite2d and SimplePlane into single Sprite?
I need an image to have perspective and some curve movement. 

![image](https://user-images.githubusercontent.com/4579539/161944347-1246f02d-ab08-44d8-bf23-11dfdbdcbc5c.png)

### Current Behavior
No idea...

- **`pixi.js` version**:  6.0.3

**Comment by ivanpopelyshev** [comment]
First , try to do your own mesh with texture , it can be SimplePlane.

Feature for curving into arbitrary projection wasnt supported enough by users so i didnt finish it. If you understand how it works - you can use, otherwise use the same method people use in other 2d engines - just make a plane that looks good enough.

**Comment by gapgag55** [comment]
SimplePlane is worked well with the curve by using mesh texture. However, I don't know how to apply the transformation into SimplePlane such as using Quad billinear from Pixi Projection. 

For example, 

```
// 1. DO MESH
const plane = new PIXI.SimplePlane(texture, rows, cols)
const { data: vertices } = plane.geometry.getBuffer('aVertexPosition');

vertices.data[i] = some change mesh....

// 2. DO PERSPECTIVE
const container = new PIXI.projection.Sprite2s(texture);
container.proj.mapBilinearSprite(container, [some quad]);
container.addChild(plane)

app.stage.addChild(container)
```
Just an idea and I'm not sure will it work. Any suggestion?

**Comment by ivanpopelyshev** [comment]
You have to make formulas for translation from screen coords to cylinder coords and back, put it to special class that i use for bilinear.

Alternatively, you can try make mesh that works in 3d with pixi-projection, and supply it with 3d coords - the result will be a bit different from SimplePlane.

It all is possible, you are one of rare users who request that feature, and it wasnt funded, nor did i have my own projects that need that. You have to do it yourself, or fund it, or wait when one of those who can do it will need it for his own projects ))

**Comment by gapgag55** [comment]
Or do you have the method that change SimplePlane into texture, so that I can transform it by Pixi-Projection

**Comment by bigtimebuddy** [comment]
For support or help related issues, please create a discussion. Thanks for helping us keep the project organized.

**Comment by gapgag55** [comment, marked as answer]
Solved, Just use "app.renderer.generateTexture()" to convert any object to texture

```
      const plane = new PIXI.SimplePlane(PIXI.Texture.from(image), 4, 4)
      const { data: vertices } = plane.geometry.getBuffer('aVertexPosition');
      vertices[5] = { x: 10, y: 10 }

      const texture = app.renderer.generateTexture(plane)

      const planWithPerspective = new Sprite2s(texture)
      planWithPerspective.proj.mapBilinearSprite(planWithPerspective, [
        { x: 0, y: 0 },
        { x: 250, y: 0 },
        { x: 250, y: 150 },
        { x: 0, y: 250 },
      ])
```
